components/movementsensor/wheeledodometry/wheeled_odometry.go

```diff
@@ -0,0 +1,302 @@
+	deps resource.Dependencies,
+	conf resource.Config,
+	logger golog.Logger,
+) (movementsensor.MovementSensor, error) {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return nil, err
+	}
+
+	timeInt := newConf.TimeInterval
+	if timeInt == 0 {
+		timeInt = defaultTimeInt
+	}
+
+	o := &odometry{
+		Named:        conf.ResourceName().AsNamed(),
+		lastLeftPos:  0,
+		lastRightPos: 0,
+		timeInterval: timeInt,
+		logger:       logger,
+	}
+
+	for i, m := range newConf.LeftMotors {
+		motorDep, err := motor.FromDependencies(deps, m)
+		if err != nil {
+			return nil, err
+		}
+		properties, err := motorDep.Properties(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+		if !properties.PositionReporting {
+			return nil, motor.NewPropertyUnsupportedError(properties, newConf.LeftMotors[i])
+		}
+		o.motors[i].left = motorDep
+	}
+
+	for i, m := range newConf.RightMotors {
+		motorDep, err := motor.FromDependencies(deps, m)
+		if err != nil {
+			return nil, err
+		}
+		properties, err := motorDep.Properties(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+		if !properties.PositionReporting {
+			return nil, motor.NewPropertyUnsupportedError(properties, newConf.RightMotors[i])
+		}
+		o.motors[i].right = motorDep
+	}
+
+	newBase, err := base.FromDependencies(deps, newConf.Base)
+	if err != nil {
+		return nil, err
+	}
+	props, err := newBase.Properties(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	o.baseWidth = props.WidthMeters
+	o.wheelCircumference = props.WheelCircumferenceMeters
+	if o.baseWidth == 0 || o.wheelCircumference == 0 {
+		return nil, errors.New("base width or wheel circumference are 0, movement sensor cannot be created")
+	}
+	o.orientation.Yaw = 0
+	o.math(context.Background())
+
+	return o, nil
+}
+
+func (o *odometry) AngularVelocity(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	return o.angularVelocity, nil
+}
+
+func (o *odometry) LinearAcceleration(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	return o.acceleration, nil
+}
+
+func (o *odometry) Orientation(ctx context.Context, extra map[string]interface{}) (spatialmath.Orientation, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	ov := &spatialmath.OrientationVector{Theta: o.orientation.Yaw, OX: 0, OY: 0, OZ: 1}
+	return ov, nil
+}
+
+func (o *odometry) CompassHeading(ctx context.Context, extra map[string]interface{}) (float64, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	return o.orientation.Yaw, nil
+}
+
+func (o *odometry) LinearVelocity(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	return o.linearVelocity, nil
+}
+
+func (o *odometry) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	return geo.NewPoint(o.position.X, o.position.Y), o.position.Z, nil
```

Comment:
I agree with what you're saying but I thought the point was that the latitude and longitude would be just local to the rover. I think that's what we really want to know. if it's at 0,0 on start up we can calculate where it's located in that frame. does that sound right @randhid

Reply:
also, I'm assuming `geo.NewPoint` creates `*geo.Point` because this is how many other movement sensors implement `Position()` and I'm assuming those work as expected

Reply:
`geo.NewPoint` definitely [does the right thing](https://github.com/kellydunn/golang-geo/blob/master/point.go#L12-L26), it's just that the point created contains latitude and longitude, rather than "meters moved forward/left since we last went through reconfiguration," and I believe that `movementsensor.Position` is intended for GPS results, just like the `geo.Point` that it returns.

Reply:
talked to rand about this and we are just going to document that it is not real world lat and long it's local to the robot and 0,0 is wherever the robot is when the server starts up